### PR TITLE
[risk=no][RW-11074] Direct users to specific Absorb courses

### PR DESCRIPTION
**Pull request by Peter-Lavigne**
Direct users to specific Absorb courses. This is a workaround since deep links appear to require support from the Absorb team. See the comments on [this ticket](https://precisionmedicineinitiative.atlassian.net/browse/RW-11074) and [this thread](https://pmi-engteam.slack.com/archives/C02MWP2RN5P/p1696878015979769) for details. I'll ask the Absorb team about deep links this week.

This only applies when Absorb is enabled for the user.

To test this, create a new user and check the data access page. The "enabledForNewUsers" flag can be toggled to see Moodle behavior (no change).

Screenshots:

Training incomplete:
image

Training complete / bypassed:
image (1)

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [x] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [x] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [x] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java [hunk 1]
```diff
@@ -111,7 +111,8 @@ public class MailServiceImpl implements MailService {
   private static final String TWO_STEP_VERIFICATION = "Turn on Google 2-Step Verification";
   private static final String CT_INSTITUTION_CHECK = "Check that %s allows Controlled Tier access";
   private static final String ERA_COMMON = "Connect your eRA Commons account";
-  private static final String RT_TRAINING = "Complete All of Us Registered Tier Training";
+  private static final String RT_TRAINING =
+      "Complete All of Us Responsible Conduct of Research Training";
```
Comment (Peter-Lavigne) [inline api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java:115]: See [this message](https://pmi-engteam.slack.com/archives/C02MWP2RN5P/p1696942677700569?thread_ts=1696878015.979769&cid=C02MWP2RN5P).

File: ui/src/app/pages/access/compliance-training-module-card-title.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+export const ComplianceTrainingModuleCardTitle = ({
+  tier,
+  profile,
+}: ComplianceTrainingModuleCardProps) => {
+  const [useAbsorb, setUseAbsorb] = React.useState(false);
+  useEffect(() => {
+    profileApi().useAbsorb().then(setUseAbsorb);
+  }, []);
```
Comment (Peter-Lavigne) [inline ui/src/app/pages/access/compliance-training-module-card-title.tsx:27]: This is inefficient and is called many times when the page is loaded, due to several renders of this component. It's difficult to change the code to only call this once.

I think this is OK for now for the following reasons:
- This call will go away once we transfer users off of Moodle entirely
- This endpoint is fairly lightweight
- This page isn't visited often
- This call will go away if we solve deep links

File: ui/src/app/pages/access/compliance-training-module-card-title.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  const { accessModule, trainingTitle, courseTitle } =
+    tier == AccessTierShortNames.Registered
+      ? {
+          accessModule: AccessModule.COMPLIANCE_TRAINING,
+          trainingTitle: 'Responsible Conduct of Research Training',
+          courseTitle: 'Responsible Conduct of Research',
```
Comment (Peter-Lavigne) [inline ui/src/app/pages/access/compliance-training-module-card-title.tsx:34]: This is different text than before. See [this comment](https://pmi-engteam.slack.com/archives/C02MWP2RN5P/p1696886542210769?thread_ts=1696878015.979769&cid=C02MWP2RN5P).

File: ui/src/app/pages/access/identity-help-text.tsx [hunk 4]
```diff
@@ -38,7 +38,7 @@ const ContactUs = (props: { profile: Profile }) => {
 
 export const IdentityHelpText = (props: {
   profile: Profile;
-  afterInitialClick: boolean;
+  afterInitialClick?: boolean;
```
Comment (Peter-Lavigne) [inline ui/src/app/pages/access/identity-help-text.tsx:41]: Many of the typing updates document pre-existing usage. One of the commits addresses this specifically.

**Review by jmthibault79 — APPROVED** [review]
Tested locally.

File: ui/src/app/pages/access/modules-for-annual-renewal.tsx [hunk 5]
```diff
@@ -581,6 +583,7 @@ export const ModulesForAnnualRenewal = (props: RenewalCardProps) => {
               radioButtonStyle={{ marginRight: '0.5em' }}
               showTimeEstimate={true}
               duccSignedVersion={profile.duccSignedVersion}
```
[the remark this answers is not in the document]
Comment (Peter-Lavigne) [reply ui/src/app/pages/access/modules-for-annual-renewal.tsx]: Nice catch, done 👍

**Pull request merged**